core/src/main/java/tachyon/worker/hierarchy/StorageDir.java

```diff
@@ -117,6 +128,7 @@ public void accessBlock(long blockId) {
     synchronized (mLastBlockAccessTimeMs) {
       if (containsBlock(blockId)) {
         mLastBlockAccessTimeMs.put(blockId, System.currentTimeMillis());
+        updateReferenceFrequency(blockId, REFERENCE_FREQUENCY_INCREMENT);
```

Comment:
When a block is added the first time, the frequency is not always the initial value. E.g. when  a block is copied from StorageDir A to StorageDir B, it should keep its previous frequency, and it's the first time added to StorageDir B.
Both an indeed new block and a copied block call the same `addBlockId(long, long, long, long, boolean)` method, so using a special flag to distinguish them.